extensions/wikia/UserLogin/js/UserLoginFacebook.js

```diff
@@ -111,10 +113,12 @@
 				// begin ajax call performance tracking
 				this.bucky.timer.start('loginCallbackAjax');
 
+				pageUrl = new QueryString();
+
 				// now check FB account (is it connected with Wikia account?)
 				$.nirvana.postJson('FacebookSignupController', 'index', {
-						returnto: encodeURIComponent(window.wgPageName),
-						returntoquery: encodeURIComponent(window.location.search.substring(1))
+						returnto: pageUrl.getVal('returnto'),
+						returntoquery: pageUrl.getVal('returntoquery')
```

Comment:
I just tested this where I was logged out, but my wikia account was already connected with facebook. I clicked on the facebook button from the global nav dropdown, was logged in, but was then redirected to the home page, even though I had started on an article page. So this is a bug.